ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java

```diff
@@ -127,6 +137,56 @@ public ReactImageView(
     mCallerContext = callerContext;
   }
 
+  public void setShouldNotifyLoadEvents(boolean shouldNotify) {
+    if (!shouldNotify) {
+      mControllerListener = null;
+      return;
+    } else {
+      final EventDispatcher mEventDispatcher = ((ReactContext) getContext()).
+          getNativeModule(UIManagerModule.class).getEventDispatcher();
+
+      mControllerListener = new BaseControllerListener<ImageInfo>() {
+        @Override
+        public void onSubmit(
+            String id,
+            Object callerContext
+        ) {
+          mEventDispatcher.dispatchEvent(
+              new ImageLoadEvent(getId(), SystemClock.uptimeMillis(), ImageLoadEvent.ON_LOAD_START)
+          );
+        }
+
+        @Override
+        public void onFinalImageSet(
+            String id,
+            @Nullable final ImageInfo imageInfo,
+            @Nullable Animatable animatable) {
+          if (imageInfo != null) {
+            mEventDispatcher.dispatchEvent(
+                new ImageLoadEvent(getId(), SystemClock.uptimeMillis(), ImageLoadEvent.ON_LOAD_END)
+            );
+            mEventDispatcher.dispatchEvent(
+                new ImageLoadEvent(getId(), SystemClock.uptimeMillis(), ImageLoadEvent.ON_LOAD)
+            );
+          }
+        }
+
+        @Override
+        public void onFailure(
+            String id,
+            Throwable throwable
+        ) {
+          mEventDispatcher.dispatchEvent(
+              new ImageLoadEvent(getId(), SystemClock.uptimeMillis(), ImageLoadEvent.ON_LOAD_END)
+          );
+        }
+      };
+    }
+
+    mIsDirty = true;
+    maybeUpdateView();
```

Comment:
This is unnecessary and may trigger additional work in case when multiple properties are updated in the same batch. This method should be only called from `onAfterUpdateTransation` which covers all the cases when properties has been modified https://github.com/facebook/react-native/blob/e4f0971a55d386db93ff5c78d383ffae1132dc33/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageManager.java#L108

(we actually also call it from `onSizeChanged` but this may happen independently, therefore it's fine)